omg/entities/player.py

```diff
@@ -85,13 +106,28 @@ def update(self, mouse_x, mouse_y, delta_time):
         )
         self._regenerate_mana(delta_time)
 
-    def shoot(self):
-        """Shoots a projectile and informs the observes with an ProjectileShotEvent."""
-        if self.skills.current_size == 0 or self.current_mana < 20:
+        # Update combined elements
+        if len(self.to_be_combined_element_cash) == 2:
+            # Get the name of the skill after combining elements
+            new_skill = "".join(self.to_be_combined_element_cash[::])
+
+            # Remove "ElementFactory" from the element name
+            new_skill = new_skill.replace("ElementFactory", "")
+
+            # Update crafted skill slots
+            self._update_crafted_skill_slots(new_skill)
+
+            # Empty the element cash
+            self.to_be_combined_element_cash = []
```

Comment:
I have couple of comments here: 

1. This logic should be encapsulated, it is complex enough to deserve its own method, which will improve readibility ( it took me some time to find where did you created the skills from elements, for example).

2. The method itself is not robust as it is very dependent to the child class names of the `ProjectileFactory`. There is no enforcing of the class name which is derived from some base class.

3. If you want to make this effective, you would not constantly pop and remove items from the list, but rather replace a specific it directly. ( this is also a comment to `_update_crafted_skill_slots` method)

4. In general, I believe the combining elements can be better structured ( this is a comment to all the changes in `projectiles.py` and to the related parts. I believe we should discuss it offline together.